test/gtest/coll/test_gatherv.cc

```diff
@@ -0,0 +1,305 @@
+                    displs[i] = all_counts;
+                    all_counts += counts[i];
+                }
+
+                coll->dst.info_v.mem_type      = mem_type;
+                coll->dst.info_v.counts        = (ucc_count_t *)counts;
+                coll->dst.info_v.displacements = (ucc_aint_t *)displs;
+                coll->dst.info_v.datatype      = dtype;
+
+                ctxs[r]->rbuf_size = ucc_dt_size(dtype) * all_counts;
+                UCC_CHECK(ucc_mc_alloc(&ctxs[r]->dst_mc_header,
+                                       ctxs[r]->rbuf_size, mem_type));
+                coll->dst.info_v.buffer = ctxs[r]->dst_mc_header->addr;
+                if (inplace) {
+                    UCC_CHECK(ucc_mc_memcpy(
+                        (void *)((ptrdiff_t)coll->dst.info_v.buffer +
+                                 displs[r] * ucc_dt_size(dtype)),
+                        ctxs[r]->init_buf,
+                        ucc_dt_size(dtype) * my_count, mem_type,
+                        UCC_MEMORY_TYPE_HOST));
+                }
+            }
+            if (r != root || !inplace) {
+                UCC_CHECK(ucc_mc_alloc(&ctxs[r]->src_mc_header,
+                                       ucc_dt_size(dtype) * my_count,
+                                       mem_type));
+                coll->src.info.buffer = ctxs[r]->src_mc_header->addr;
+                UCC_CHECK(ucc_mc_memcpy(coll->src.info.buffer,
+                                        ctxs[r]->init_buf,
+                                        ucc_dt_size(dtype) * my_count,
+                                        mem_type, UCC_MEMORY_TYPE_HOST));
+            }
+            if (inplace) {
+                coll->mask  |= UCC_COLL_ARGS_FIELD_FLAGS;
+                coll->flags |= UCC_COLL_ARGS_FLAG_IN_PLACE;
+            }
+            if (persistent) {
+                coll->mask  |= UCC_COLL_ARGS_FIELD_FLAGS;
+                coll->flags |= UCC_COLL_ARGS_FLAG_PERSISTENT;
+            }
+        }
+    }
+    void data_fini(UccCollCtxVec ctxs)
+    {
+        for (auto r = 0; r < ctxs.size(); r++) {
+            ucc_coll_args_t *coll = ctxs[r]->args;
+            if (r == root) {
+                UCC_CHECK(ucc_mc_free(ctxs[r]->dst_mc_header));
+                free(coll->dst.info_v.counts);
+                free(coll->dst.info_v.displacements);
+            }
+            if (r != root || !inplace) {
+                UCC_CHECK(ucc_mc_free(ctxs[r]->src_mc_header));
+            }
+            ucc_free(ctxs[r]->init_buf);
+            free(coll);
+            free(ctxs[r]);
+        }
+        ctxs.clear();
+    }
+    void reset(UccCollCtxVec ctxs)
+    {
+        ucc_coll_args_t *coll     = ctxs[root]->args;
+        size_t           my_count = coll->src.info.count;
+        ucc_datatype_t   dtype    = coll->dst.info_v.datatype;
+        int *            displs   = (int *)coll->dst.info_v.displacements;
+
+        clear_buffer(coll->dst.info_v.buffer, ctxs[root]->rbuf_size,
```

Comment:
how clear_buffer works for non root ranks?

Reply:
coll is only for root, line 119: ucc_coll_args_t *coll     = ctxs[root]->args;